src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -337,7 +337,10 @@ private function createTypeAlternatives(TypedReference $reference)
             return ' '.$message;
         }
 
-        if (isset($this->ambiguousServiceTypes[$type])) {
+        $servicesAndAliases = $this->container->getServiceIds();
+        if (!$this->container->has($type) && false !== $key = array_search(strtolower($type), array_map('strtolower', $servicesAndAliases))) {
+            return sprintf(' Did you mean "%s"?', $servicesAndAliases[$key], $type);
```

Comment:
Once placeholder but two arguments for `sprintf()`?

Reply:
@sroze could you please remove $type ?
After that I think the PR is mergeable ;).

Reply:
Oh, good point, well spotted 👍

Reply:
Updated